src/entities/road.lua

```diff
@@ -86,10 +86,39 @@ function road:draw()
   love.graphics.setColor(255, 255, 255, 255)
 end
 
-function road:update(angle, roadShift)
+  
+function road:update(newSegments)
   -- create a road segment at the position and angle of the frontier.
-  self:addsegment()
+  for _, s in pairs(newSegments) do
+  self:addSegment(s.x, s.y, s.angle)
+  end
+  i = table.getn(newSegments)
+  self:setFrontier(newSegments[i].x, newSegments[i].y, newSegments[i].angle, "center")
```

Comment:
we don't need to call `self:setFrontier()` here because main has already called `road:pushFrontier()` for us, which has already called it for us.

Reply:
This is only true for the client that actually calls `road:pushFrontier()` the road. My thought was that once `newSegments` was built for the client that touched the frontier, `newSegments` would be sent out to all the other clients who would call `road:update()`

Reply:
ah okay!